README.md

```diff
@@ -268,42 +354,170 @@ the_data = { source = "../data-2017-02-05", copy_git_ignored = true }
-    Some(p) => p.join("the_data"),
-    None => path::PathBuf::from("../data-2017-02-05"),
-}
+### Package filtering
+
+Rust workspace builds all contained packages by default. However, when a worspace project builds multiple platforms, some packages might not be buildable on all targets. To help with that situation, Dinghy has a filter option which allows benching, building or testing only the packages compatible with the current target platform.
+
+To make it works, modify the package toml to include some Dinghy metadata specifying the allowed rustc triples. For example, here we allow android only:
+```toml
+[package.metadata.dinghy]
+allowed_rustc_triples = [ "armv7-linux-androideabi", "aarch64-linux-android", "i686-linux-android", "x86_64-linux-android" ]
+```
+
+Or specifying the rustc triples to ignore. For example, here we disallow ios:
+```toml
+[package.metadata.dinghy]
+ignored_rustc_triples = ["aarch64-apple-ios", "armv7-apple-ios", "armv7s-apple-ios", "i386-apple-ios", "x86_64-apple-ios"]
+```
+
+### Environment variables
+
+Dinghy allows defining environment variables per-platform. These variables will be set during the whole build process (including build.rs scripts targeting either the host or target platforms).
+
+```toml
+[platforms.my-platform]
+env={ MY_ENV="my-value" }
+```
+
+Here is an example environment variable that helps running openssl build script for the host platform (when another dependency build script uses openssl to download some stuff) during an android cross compilation build:
+```toml
+[platforms.android-arm64]
+env={ X86_64_UNKNOWN_LINUX_GNU_OPENSSL_DIR = "/usr" } 
+```
+
+It's possible to setup environment variables for a (non cross-compilation) build running for the host platform too:
+```toml
+[platforms.host]
+env={ MY_ENV="my-value" }
+```
+
+### Overlays
+
+A toolchain might not contains all the required dependencies for your project. To help with situation, Dinghy offers overlays.
+
+#### Overlay configuration
+
+By default, Dinghy will look for overlays in the `dinghy/overlay/<platform>` directory next to your configuration file.
+
+Overlay directories can also be specified using the overlays section of your platform configuration:
+```toml
+[platforms.android-arm64]
+overlays={ mydep={ path="/mypath" } } 
+```
+
+#### Overlay directory
 
-let the_conf_file = match test_data_path() {
-    Some(p) => p.join("conf_file"),
-    None => path::PathBuf::from("/etc/some/file"),
+An overlay is a directory which contains the required *.so*, *.h* and *.pc* files for a dependency. For example:
 ```
+my-overlay
+|- libmylib.so
+|- bmylib.h
+|- bmylib.pc
+```
+
+An overlay is like an additional sysroot. So you can also create directories and subdirectories. For example, we use a tensorflow overlay on our android projects with the following structure:
+```
+android-arm64
+|- overlay
+    |- tensorflow
+        |- usr
+            |- include
+                |- pkgconfig
+                    |- mylib.pc
```

Comment:
mylib when saying it's a tensorflow overlay

Reply:
k